frontend/README.md

```diff
@@ -0,0 +1,93 @@
+# Abstract Account App
+
+## Introduction
+
+The Abstract Account App is a Next.js-based application that interacts with the
+Abstract Account system on the blockchain. It enables users to connect their
+wallets, create and manage Abstract Accounts, query account information, and
+interact with smart contracts in a type-safe manner using Abstract Codegen.
+
+## Features
+
+- Wallet connection and management using Graz
+- Creation of Abstract Accounts
+- Querying Abstract Account information via the Abstract Subgraph GraphQL API
+- Code generation for type-safe interactions with smart contracts using Abstract
+  Codegen
+
+## Prerequisites
+
+Before you begin, ensure you have met the following requirements:
+
+- pnpm
+- A compatible wallet (e.g., Keplr for Cosmos-based chains)
+
+## Installation
+
+1. Clone the repository:
+   ```
+   git clone git@github.com:AbstractSDK/templates.git
+   ```
+
+2. Navigate to the project directory:
+   ```
+   cd templates/frontend
+   ```
+
+3. Install the dependencies:
+   ```
+   pnpm install
+   ```
+
+## Configuration
+
+1. Create a `.env` file in the root directory and add the following:
+   ```
+   ABSTRACT_SUBGRAPH_URL=https://api.abstract.money/graphql
+   ```
+
+## Code Generation
+
+This project uses code generation for type-safe interactions with smart
+contracts and GraphQL queries. To generate the necessary code, run:
+
+```
+pnpm generate
+```
+
+This command will execute the following scripts:
+
+1. `generate:graz`: Generates chain information for Graz
+2. `generate:gql`: Generates TypeScript types for GraphQL queries using the
+   GraphQL Code Generator
+3. `generate:abstract`: Generates TypeScript types, methods, and hooks for smart
+   contract interactions using Abstract CLI
+
```

Comment:
We should add a section on updating `abstract.config.js` with the proper module name, namespace, and version, and note that the codegen command referencing module-generated code *must* be run after the module schema generation.